### DISCUSSION TITLE
How to Automatically Generate Columns in DataGrid with Spaced-Out Column Names?

### DISCUSSION THREAD

**Discussion by al-zakaria** [AvaloniaUI/Avalonia#17599] [Q&A]
Hi! I’d like to know how to set up automatic column generation in DataGrid so that multi-word column names are split up for better readability. For example, I'd like LastName to appear as "Last Name" and Email as "E-mail". Is there any built-in support for this in Avalonia UI?

I've written my own code to handle it, but it’s quite large and complicated. I’m wondering if there’s a simpler, built-in way to achieve this.

**Comment by timunie** [comment]
Not that I know. There's an autogenerating event which may help but the word correction is your task I think.

**Reply by robloo** [reply]
Yes, the requested functionality is highly application-specific and culture/language dependent. We can't do this type of thing in the framework itself. You will have to handle this in your application as you are already doing.
